core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -188,9 +196,9 @@ void OpenMPInternal::initialize(int thread_count) {
     // Before any other call to OMP query the maximum number of threads
     // and save the value for re-initialization unit testing.
 
-    Impl::g_openmp_hardware_max_threads = get_current_max_threads();
+    g_openmp_hardware_max_threads = get_current_max_threads();
 
-    int process_num_threads = Impl::g_openmp_hardware_max_threads;
+    int process_num_threads = g_openmp_hardware_max_threads;
```

Comment:
Wouldn't it make sense to only set this for the default execution space instance?

Reply:
I think so.  But such a refactor/fix is beyond the scope of this PR.